### ISSUE TITLE
Amsmath's \big and text font change

### ISSUE THREAD

**Issue by blefloch** [latex3/latex2e#49]
## Brief outline of the bug

When amsmath is in use, `\big` in text mode uses the size of the last math formula.  Note that the command `\big` is valid in text mode (for instance Knuth suggests it for text parentheses when the contents contains a tall math expression).

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}
\documentclass{article}
\usepackage{amsmath}
\begin{document}
The number $\varphi$ \big(equal to $(1+\sqrt{5})/2$\marginpar{\tiny Golden ratio~$\varphi$.}\big) is nice.
\end{document}
```

## Log file (required) and possibly PDF file  
[test.log](https://github.com/latex3/latex2e/files/1997886/test.log)
[test.pdf](https://github.com/latex3/latex2e/files/1997887/test.pdf)

**Comment by davidcarlisle** [comment]
Something like this I guess? (for both issues)

[\documentclass{article}

\makeatletter
\def\big#1{{\leavevmode
   \hbox{$\left#1\vbox to8.5\p@{}\right.\n@space$}}}

\usepackage{amsmath}

\def\bBigg@#1#2{%
  {\leavevmode\check@mathfonts\@mathmeasure\z@{\nulldelimiterspace\z@}%
     {\left#2\vcenter to#1\big@size{}\right.}%
   \box\z@}}

\begin{document}
We claim that $16^{3^2} = (8^4)^3$.

\big(Recall that $x^{y^z}=x^{(y^z)}$.\big)




The number $\varphi$ \big(equal to $(1+\sqrt{5})/2$\marginpar{\tiny Golden ratio~$\varphi$.}\big) is nice.

\end{document}](url)

**Comment by davidcarlisle** [comment]
Actually might be better to put the `\check@mathfonts` inside the `\hbox` in the definition of `\@mathmeasure`

**Comment by blefloch** [comment]
Would it be better for `\leavevmode` to be outside the group, to avoid possible issues with `\everypar` being run in a group?  I would expect something like `\ifvmode\leavevmode\fi{....}` to be better.  On the other hand `\mbox` does `{\leavevmode\hbox{...}}` so I suppose it's safe enough.

**Comment by davidcarlisle** [comment]
@blefloch  well I was worried about `$x^\big($` which is pretty crazy markup but you never know....  also there may be other more reasonable cases that are implicitly grabbing a math `{}` group.

**Comment by blefloch** [comment]
> |$x^\big($| which is pretty crazy markup but you never know....

This can only happen in math mode so \ifvmode\leavevmode\fi should work.

I suggest sticking to what work in \mbox (or even to just replacing
\hbox by \mbox if that's simpler, I don't know).

**Comment by blefloch** [comment]
At #50 Enrico points out a few other commands that don't have `\leavevmode`, and if I understand him properly he suggests that that's fine because they are not documented in the LaTeX manual.

I would argue that at least `\smash` should have `\leavevmode` since it explicitly distinguishes math mode from non-math mode.  Now it's possible actually that people want the vmode behaviour for `\smash`?

**Comment by FrankMittelbach** [comment]
One of the cornerstones of LaTeX behavior is that everything that produces or acts on textual material ends up being or starting LR mode, i.e., Leslie tried hard to add \leavevmode in all such places to avoid the behavior of, say, \hbox that acts differently if at the start of paragraph (ie still being in vmode) or in the middle of it.

The downside of this approach is that sometimes you find yourself getting into LR mode when you really don't want it: good example is a tabular* or minipage of full \textwidth which gets an \indent added in front while one really want to have something filling the whole line.

However, because of the consisteny here it is understandable what happens and how to avoid it.

I therefore think that the correct LaTeX approach for \smash and \phantom is in fact starting LR mode including (implicitly adding the indentation if in vmode) and not staying in vmode, ie behave like mbox in that respect.

**Comment by blefloch** [comment]
For the record, I was wrong about Knuth advising to use \big in hmode.
In fact in the TeXbook's source he uses $\bigl($...$\bigr)$ in a couple
of places, but not directly \big(.  Thanks Karl for making me look more
carefully.

I still think it would be good to make \big(...\big) work properly, but
I can understand if people don't want to.

**Comment by FrankMittelbach** [comment]
Actually in amsmath `\big` doesn't work properly outside math mode and I'm not sure it is worth trying to fix as that would slow down all use of that code in documents. Of course that then begs the question if it is the right choice to allow `\big` in std LaTeX outside math or make those math only. For `\phantom` and `\smash` it is clear that they can and are used in text, but for `\big(ggg)` that is not so clear.

Here is an MWE with the problem:
```
\documentclass{article}

\usepackage{amsmath}

\begin{document}

This is \big( no good \big)!

\setbox0\hbox{$ $}

But this \big( is \big).

\end{document}

```
Basically amsmath circumvents the font inits (under the assumption that `\@mathmeasure` is only called in math mode) and so you may end up with unset math fonts if this is the first time that math is used.

**Comment by car222222** [comment]
I have never, ever before this seen or heard any reason for using \big outside mathmode.  I know of no traditional typesetting use of distinct sizes of parentheses etc. in text (but we can always invent such a tradition).

If there is a use-case for such control of the size of parenthetical markers in text, then write a package that robustly provides what is needed, and use diffferent command names.

**Comment by blefloch** [comment]
I am not the only one to want bigger parentheses in text mode when the
text contains maths that also has parentheses:
https://tex.stackexchange.com/questions/249676/size-of-parenthesis-in-text-mode

**Comment by car222222** [comment]
Whilst, from a typographic viewpoint, I would question the aesthetics of distinguishing these by making the text parentheses larger than normal, this (and presumably Bruno’s own detailed requirements) certainly show a need for such a distinction.  However, it would be better to provide this possibility (together with smaller versions) with new text-mode commands.

**Comment by eg9** [comment]
> On 29 Sep 2018, at 23:41 , Chris Rowley <notifications@github.com> wrote:
> 
> Whilst, from a typographic viewpoint, I would question the aesthetics of distinguishing these by making the text parentheses larger than normal, this (and presumably Bruno’s own detailed requirements) certainly show a need for such a distinction. However, it would be better to provide this possibility (together with smaller versions) with new text-mode commands.

I fully agree. We need to preserve older documents that might
have used `\big` outside of math mode (but most likely in
horizontal mode); on the other hand, we can provide well
defined commands for those who seem to need them.

Something like \textbiglpar and \textbigrpar or whatever
is decided upon.

Advertising \big( in text mode is really bad, in my opinion.


Ciao
Enrico

**Comment by blefloch** [comment]
On 09/29/2018 11:56 PM, Enrico Gregorio wrote:
> 
>> On 29 Sep 2018, at 23:41 , Chris Rowley <notifications@github.com> wrote:
>>
>> Whilst, from a typographic viewpoint, I would question the aesthetics
> of distinguishing these by making the text parentheses larger than
> normal, this (and presumably Bruno’s own detailed requirements)
> certainly show a need for such a distinction. However, it would be
> better to provide this possibility (together with smaller versions) with
> new text-mode commands.
> 
> I fully agree. We need to preserve older documents that might
> have used `\big` outside of math mode (but most likely in
> horizontal mode); on the other hand, we can provide well
> defined commands for those who seem to need them.

I certainly have used it in documents that are on the arxiv and
published in some journal with TeX-based workflow.  Since I distinctly
remember reading the advice somewhere, it is very likely that others
have done the same.

> Something like \textbiglpar and \textbigrpar or whatever
> is decided upon.
> 
> Advertising \big( in text mode is really bad, in my opinion.

I don't yet see why it should be so bad.  There are other commands that
work in both text and math mode, e.g., various spaces, \dots

**Comment by FrankMittelbach** [comment]
Am 30.09.18 um 09:36 schrieb Bruno Le Floch:
> 
> I don't yet see why it should be so bad. There are other commands that
> work in both text and math mode, e.g., various spaces, \dots

what is potentially bad about is that you get a "math" parentheses which 
may not at all be the same as the corresponding character from the text 
font. For Computer Modern it doesn't make a difference, but try it with 
other fonts and the results may not look great. It is a bit like writing
```
   possible values are 1 and $-1$
```
works great most of the time but try it with say Concrete Roman + Euler 
Math (aka beton package)

**Comment by blefloch** [comment]
> what is potentially bad about is that you get a "math" parentheses which
> may not at all be the same as the corresponding character from the text
> font.

Good point, thanks.

**Comment by car222222** [comment]
Is there not already a package that provides relative text sizes?  Maybe even something like 
\textbigger {(}  ??

We did at some stage think of adding such to the 2e kernel.

**Comment by davidcarlisle** [comment]
On Sun, 30 Sep 2018 at 11:01, Chris Rowley <notifications@github.com> wrote:

> Is there not already a package that provides relative text sizes? Maybe
> even something like
> \textbigger {(} ??
>
> We did at some stage think of adding such to the 2e kernel.
>
>
>
there is Donald Arseneau's relsize, although arguably you really want
stretchy brackets here rather than a font size change.

David

**Comment by car222222** [comment]
In text!  ?? 
> stretchy brackets here rather than a font size change.

That would surely encourage people to add tables, etc., etc. straight into text!

**Comment by car222222** [comment]
But these days, to get that sophisticated glyph in a text font you would need a Unicode slots for BigParen, etc.

So . . . Let’s stick to animated smilies:-).

> I can't say I have used \big( in text at all, but if I did use it then if
> (as opentype fonts are likely to have) the font has a variant slightly
> taller but not fatter ( then that might be better big that simply using a
> scaled version of the font.

**Comment by davidcarlisle** [comment]
On Sun, 30 Sep 2018 at 13:46, Chris Rowley <notifications@github.com> wrote:

> But these days, to get that sophisticated glyph in a text font you would
> need a Unicode slots for BigParen, etc.
>

in luatex and xetex you can access variant glyphs in the same slot (and
that's what the math \big  will do essentially as it just gets whatever
variants are lined up in the font metrics without changing fonts, possibly
only math fonts have variant parens currently but that needn't be the case.

David
